### PR TITLE
[DOCS] Defines data frame transform resources

### PR DESCRIPTION
**Pull request by lcawl**
The "Pivot resources" page (https://www.elastic.co/guide/en/elasticsearch/reference/master/data-frame-transform-pivot.html) defines objects that are used in the data frame APIs (https://www.elastic.co/guide/en/elasticsearch/reference/master/data-frame-apis.html).

It is missing information about destination objects, source objects, and the larger concept of data frame transform resources. This PR rectifies those omissions.

### CODE REVIEW

**Comment by lcawl** [comment]
You can preview these changes here: https://doc-review-63bf6.firebaseapp.com/es3/data-frame-transform.html

File: docs/reference/data-frames/apis/get-transform.asciidoc [hunk 1]
```diff
@@ -88,7 +88,7 @@ are no matches or only partial matches.
 
 `transforms`::
   (array) An array of transform resources, which are sorted by the `id` value in
-  ascending order.
+  ascending order. See <<data-frame-transform>>.
```
Comment (szabosteve) [inline docs/reference/data-frames/apis/get-transform.asciidoc]: In line with the suggested change below.

```suggestion
  ascending order. See <<data-frame-transform-resource>>.
```

File: docs/reference/rest-api/defs.asciidoc [hunk 2]
```diff
@@ -8,22 +8,22 @@ These resource definitions are used in APIs related to {ml-features} and
 * <<ml-calendar-resource,Calendars>>
 * <<ml-datafeed-resource,{dfeeds-cap}>>
 * <<ml-datafeed-counts,{dfeed-cap} counts>>
+* <<data-frame-transform,{dataframe-transforms-cap}>>
```
Comment (szabosteve) [inline docs/reference/rest-api/defs.asciidoc]: In line with the suggested change above.

```suggestion
* <<data-frame-transform-resource,{dataframe-transforms-cap}>>
```

**Review by szabosteve — COMMENTED** [review]
I had a couple of minor comments.
It would be nice to agree on the naming convention of the resource documentation IDs. If we can find the solution that we want to follow, we should change the names that are not in line with the decision.

**Review by sophiec20 — APPROVED** [review]
LGTM, with some observations.

1. We always tend to show "group_by" first and "aggregations" below.  I assume we are listing alphabetically in the resource, which is ok providing we can show the examples with the `group_by` at the top.

2. `aggs` is a supported short form for `aggregations` - do we ever mention that?

3. "Examples" as a heading has less prominence than "Dest Object" and the other objects, and it's a bit strange how it sits between top level properties and the object definitions. This seems odd but may just be a firebase formatting thing.

**Comment by lcawl** [comment]
> * We always tend to show "group_by" first and "aggregations" below.  I assume we are listing alphabetically in the resource, which is ok providing we can show the examples with the `group_by` at the top.

That's right, they're just sorted alphabetically in these definition lists.

> * `aggs` is a supported short form for `aggregations` - do we ever mention that?

I've added that , thanks!

> * "Examples" as a heading has less prominence than "Dest Object" and the other objects, and it's a bit strange how it sits between top level properties and the object definitions. This seems odd but may just be a firebase formatting thing.

I've moved the "Examples" section to the bottom of the page since you're right it applies to all the objects too.

**Pull request merged**